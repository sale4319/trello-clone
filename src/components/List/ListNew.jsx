import React, { useState, useEffect } from 'react';
import { useCreateList } from '../../api/apiHooks/apiLists';

const ListNew = () => {
    const [isActive, setIsActive] = useState(false);
    const [nameField, setNameField] = useState('');
    const [newListName, setNewListName] = useState('');
    const { isSuccess } = useCreateList(newListName);

    useEffect(() => {
        if (isSuccess && newListName) {
            setNewListName('');
            setNameField('');
            setIsActive(false);
        }
    }, [isSuccess, newListName]);

    const handleKeyboardPress = (event) => {
        if (event.key === 'Enter') {
            setNewListName(nameField);
        }
    };

    return (
        <div className="list-new-container">
            {isActive ? (
                <input
                    value={nameField}
                    onChange={event => setNameField(event.target.value)}
                    onKeyPress={event => handleKeyboardPress(event)}
                    onBlur={() => setIsActive(false)}
                    autoFocus
                />
            ) : (
                <div
                    className="list-new-button"
                    onClick={() => {
                        setIsActive(true);
                    }}>
                    + Add another list
                </div>
            )}
        </div>
    );
};

export default ListNew;