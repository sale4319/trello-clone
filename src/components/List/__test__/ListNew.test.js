import React from 'react';
import  ListNew  from '../ListNew';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

test('component renders and takes snapshot', () => {
    const component = render(<ListNew/>);
    expect(component.container).toMatchSnapshot();
  });