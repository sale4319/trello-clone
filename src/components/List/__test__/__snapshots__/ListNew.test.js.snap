// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders and takes snapshot 1`] = `
<div>
  <div
    class="list-new-container"
  >
    <div
      class="list-new-button"
    >
      + Add another list
    </div>
  </div>
</div>
`;
